docs/documentation/development/bindings/firmware-update.md

```diff
@@ -0,0 +1,118 @@
+---
+layout: documentation
+---
+
+{% include base.html %}
+
+# Implementing a Firmware Update Handler
+
+Bindings can implement the `org.eclipse.smarthome.core.thing.firmware.FirmwareUpdateHandler` interface in order that the firmware for the physical devices of their things can be updated. If this interface is implemented by the corresponding `ThingHandler` implementation then the framework will take care of registering the `FirmwareUpdateHandler` automatically as an OSGi service.
+
+The `FirmwareUpdateHandler` consists of the following operations:
+* `getThing()`: Returns the thing that is handled by the firmware update handler. 
+* `updateFirmware(org.eclipse.smarthome.core.thing.firmware.Firmware firmware, org.eclipse.smarthome.core.thing.firmware.ProgressCallback progressCallback)`: Updates the firmware for the physical device of the thing that is handled by the firmware update handler. The given progress callback can be used to notify the framework of a new progress step that is going to be executed.
+* `isUpdateExecutable()`: Returns true if the firmware update handler is in a state in which the firmware update for the device can be executed,
```

Comment:
Please correct me if I'm wrong, but I think in this context, 'executable' means that it is possible to perform an firmware update, and not that the firmware update is ready to be executed/started? I think this should be made clearer.